pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -98,7 +105,7 @@ def reset(self) -> None:
         # track epoch output
         self._epoch_output = [[] for _ in range(self.batch_loop.num_active_optimizers(self.total_batch_idx))]
 
-        if not self.restarting or self._num_training_batches_reached():
```

Comment:
I ran your branch and I see the same issue here again #8835

Reply:
Interesting. This might be problematic as it would reset the progress even if validation hasn't been run. I need to find about the right solution there.

Reply:
we could save to the state_dict if validation has run or not

Reply:
Yes, this is what I ended up doing.